packages/core/contracts/zodiac/OptimisticOracleModule.sol

```diff
@@ -0,0 +1,232 @@
+    }
+
+    function executeProposal(uint256 _proposalId, uint256 _transactionIndex) public payable nonReentrant() {
+        // execute transactions in an approved proposal using exec() function
+        Proposal storage proposal = proposals[_proposalId];
+
+        // this will revert if the price has not settled
+        int256 price = _getOracle().getPrice(identifier, proposal.requestTime, proposal.ancillaryData);
+
+        Transaction memory transaction = proposal.transactions[_transactionIndex];
+
+        require(
+            _transactionIndex == 0 || proposal.transactions[_transactionIndex - 1].to == address(0),
+            "Previous tx not yet executed"
+        );
+        require(transaction.to != address(0), "Tx already executed");
+        require(price != 0, "Proposal was rejected");
+        require(msg.value == transaction.value, "Must send exact amount of ETH");
+
+        // Delete the transaction before execution to avoid any potential re-entrancy issues.
+        delete proposal.transactions[_transactionIndex];
+
+        require(
+            exec(transaction.to, transaction.value, transaction.data, transaction.operation),
+            "Failed to execute the transaction"
+        );
+        emit TransactionExecuted(_proposalId, _transactionIndex);
+    }
+
+    function deleteProposal(uint256 _proposalId) public onlyOwner {
+        // delete a proposal that governance decided not to execute
+        delete proposals[_proposalId];
+        emit ProposalDeleted(_proposalId);
+    }
+
+    function deleteRejectedProposal(uint256 _proposalId) public {
+        // execute transactions in an approved proposal using exec() function
+        Proposal storage proposal = proposals[_proposalId];
+
+        // this will revert if the price has not settled
+        int256 price = _getOracle().getPrice(identifier, proposal.requestTime, proposal.ancillaryData);
+
+        // check that proposal was rejected
+        require(price == 0, "Proposal was not rejected");
+
+        // Delete the proposal
+        delete proposals[_proposalId];
+        emit ProposalDeleted(_proposalId);
+    }
+
+    function _getOptimisticOracle() private view returns (SkinnyOptimisticOracleInterface) {
+        return
+            SkinnyOptimisticOracleInterface(finder.getImplementationAddress(OracleInterfaces.SkinnyOptimisticOracle));
+    }
+
+    function _getOracle() private view returns (OracleAncillaryInterface) {
+        return OracleAncillaryInterface(finder.getImplementationAddress(OracleInterfaces.Oracle));
+    }
+
+    function _isContract(address addr) private view returns (bool isContract) {
+        return addr.code.length > 0;
+    }
+
+    function _getCollateralWhitelist() internal view returns (AddressWhitelistInterface) {
+        return AddressWhitelistInterface(finder.getImplementationAddress(OracleInterfaces.CollateralWhitelist));
+    }
+
+    function _sync() internal {
+        skinnyOptimisticOracle = _getOptimisticOracle();
```

Comment:
I dont think this needs to be internal seeing you only do it one place. also let's be a bit more descriptive with the function name. perhaps `syncUmaEcosystemContracts`. `sync` like you have it here has implications around balance synchronisations (like in Uniswap) that is not done here.